### DISCUSSION TITLE
.NET 8.0 -- Across the Board...

### DISCUSSION THREAD

**Discussion by davidcorbin-atmosera** [CommunityToolkit/Windows#380] [Q&A]
So many things are still pointing to .NET 6.0 and there are breaking changes (in many ways) to .NET 8.0 --

Is there a curated collection of links (source, documents, samples, et. al.) that *only* contains material that is ready to go with .NET 8.0 ??

**Comment by michael-hawker** [comment]
Hi @davidcorbin-atmosera can you point out some examples of what you're referring to?

We're in the process of making a release of the Toolkit which has 8.0 as a supported target. There were breaking changes to RID handling made in .NET 8, so it took us a bit to update our bit.

**Reply by davidcorbin-atmosera** [reply]
Thanks for the response. After hours here [EST - Florida]. I could not get any of the samples to run. So I pulled the Source Code, and even GenerateAllSolutions fails.  At the end of the work day, I had pulled down the code for https://github.com/microsoft/slngen/ and was just starting to get into that (because it also will not run on a machine with *just* the latest Win11, VS-2022, and .NET 8.0.  Even the Upgrade tool was not sufficient.

If you want Ican (when I get back to my work environment in the morning) post more details.  The RID issue I was aware of [as per https://learn.microsoft.com/en-us/dotnet/core/compatibility/sdk/8.0/rid-graph] but it does not see to end there... 

slngen seems to be upset with some of the aspects of some of the project file formats..  I do have with me a note that the error messages were of the form:
 error MSB4184: The expression "[System.IO.Path]::GetDirectoryName('')" cannot be evaluated. The path is not of a legal form.

**Reply by davidcorbin-atmosera** [reply]
**UPDATE: This may be a red-herring...]**
OK, a bit of a lead....

In the file:
`CommunityToolKit\Windows\tooling\ProjectHeads\AllComponents\Tests.Uwp\CommunityToolkit.Tests.Uwp.csproj`

There is a reference to:
`  <Import Project="$(MSBuildExtensionsPath)\Microsoft\WindowsXaml\v$(VisualStudioVersion)\Microsoft.Windows.UI.Xaml.CSharp.targets" />`

Debugging SlnGen shows that the $(MSBuildExtensionsPath)\Microsoft is good, but there is NO WindowsXaml sub-directory there...

So
`{"Could not find a part of the path 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\BuildTools\\MSBuild\\Microsoft\\WindowsXaml\\v17.0\\Microsoft.Windows.UI.Xaml.CSharp.targets'."}`

is an inevitable failure.....

However, I do seem to have all of the workloads for VS-2022.... so further investigation continues....

**Comment by davidcorbin-atmosera** [comment]
OK, though the above was throwing exceptions (and I see nothing exceptional about this environment)... it looks like the original error message results from

`    <_ExtrasSdkBaseDirectory>$([System.IO.Path]::GetDirectoryName($(**ToolDepsJsonGeneratorProject**)))\..\..\..\</_ExtrasSdkBaseDirectory>`

in 
`.nuget\packages\msbuild.sdk.extras\3.0.23\Sdk`  

at Line #37....   

So continuing to chase the rabbit....

**Reply by davidcorbin-atmosera** [reply]
OK, a little more isolation [stince I can now debug inside of SlnGen :) 

`D:\repos\GitHub\CommunityToolKit\Windows\components\Segmented\src\CommunityToolkit.WinUI.Controls.Segmented.csproj`

is an older project and attempting to open in VS-2022 [17.8.4] says it requires an upgade.... but when loaded via MsBuild it triggers the loadingof Older NuGet packages which later result in the msbuild.sdk.extras being (_apparently_) an older version than the actual SDK version so things go south from there...

At least that is my analysis so far...

**Reply by michael-hawker** [reply]
Thanks for digging into this @davidcorbin-atmosera, appreciate the efforts. It'd be good to understand what's happening here, as we haven't seen this type of environment issue elsewhere. I've just setup a new devbox myself and repulled these projects and haven't encountered an issue yet.

FYI @Arlodotexe as he's also been working on updating our Wiki for the project build requirements: https://github.com/CommunityToolkit/Windows/wiki/Quick-Start

If something is missing here, it'd be good to know. Generally, running the bat file from the VS Developer Prompt should just install slngen and open the solution with the dotnet tools command when the proper workloads and Windows SDK are installed.

**Reply by michael-hawker** [reply]
Actually, I had forgotten about this issue, just filed with details here, but I don't think this sounds the same as what you've been hitting: https://github.com/CommunityToolkit/Tooling-Windows-Submodule/issues/183
